src/patternfly/sass-utilities/mixins.scss

```diff
@@ -29,13 +29,39 @@
   }
 }
 
-// Creates flex item
-@mixin pf-utility-builder($class-name) {
-  // generate column span modifiers
-  @each $breakpoint, $breakpoint-value in $pf-global--breakpoint-map {
-    .pf-u-#{$class-name}#{$breakpoint-value} {
-      @include pf-media-query($breakpoint) {
-        @content;
+/// Create single prop / value classes, optionally add responsive suffix
+/// @group mixins
+/// @moduleType mixin
+/// @parameter: {Suffix} xs, sm, md, lg, xl, base or null
+/// @usage: @include pf-utility-builder(class-name, base sm md lg xl);
+/// ===============================================================================================
+/// non-responsive, base only                @include pf-utility-builder(class-name) { prop: val !important; }
+/// responsive, not including base           @include pf-utility-builder(class-name, sm lg) { prop: val !important; }
+/// responsive, including base               @include pf-utility-builder(class-name, base sm lg) { prop: val !important; }
+/// responsive, including all breakpoints    @include pf-utility-builder(class-name, $pf-global--breakpoint-list) { prop: val !important; }
+/// ===============================================================================================
+@mixin pf-utility-builder($class-name, $breakpoints: null) {
+  // if only class-name is passed to mixin, with no breakpoint params, return the prop / value declaration
+  @if $breakpoints == null or $breakpoints == "base" {
```

Comment:
Can you verify that the if/else block is necessary? I tested it locally and looks like the `@else` condition should handle it. The `@each` block will loop with the default `null` value, and set an empty `$suffix`, if you don't pass `$breakpoints` in `pf-utility-builder()`. And the `@each` block will also handle `"base"`.

Reply:
You're correct, it's a redundancy meant to circumvent the `media-query()` mixin altogether. Rationale being it's easier to read/maintain that way. Also correct that the `media-query()` mixin will handle the case you mentioned.

Reply:
I think you're right though, your suggestion would make it more efficient